packages/perseus/src/widgets/interactive-graphs/graphs/components/axis-tick-labels.tsx

```diff
@@ -0,0 +1,236 @@
+    graphEdge,
+    position,
+}: ShowTickLabelProps): boolean => {
+    let showLabel = true;
+
+    // We only want to show the initial negative tick labels (e.g. -1) on each
+    // axis if the tickStep > gridStep, to ensure that these labels do not overlap.
+    // e.g. If gridStep = 1 and tickStep = 2, there are 2 grid lines for every 1 tick,
+    // which allows enough room for both these tick labels to render.
+    showLabel = tickStep > gridStep ? true : label !== -tickStep;
+
+    // If the tick label position is greater than the relevant edge of the graph, don't render it
+    if (position <= -graphEdge || position >= graphEdge) {
+        showLabel = false;
+    }
+
+    return showLabel;
+};
+
+const YGridLabel = ({label, gridStep, tickStep, graphInfo}: GridLabel) => {
+    const pointOnAxis: vec.Vector2 = [0, label];
+    const pixelPoint = pointToPixel(pointOnAxis, graphInfo);
+
+    // If the axis is outside of the graphs viewbox/range we want to make sure
+    // the labels are still visible by clamping them to the edge of the graph
+    const leftPadding = 40;
+    const left = Math.min(
+        Math.max(pixelPoint[0] - leftPadding - tickPadding, -leftPadding),
+        graphInfo.width + 5,
+    );
+
+    // Provide a little padding above the tick label
+    const topPadding = 7;
+    const top = pixelPoint[1] - topPadding;
+
+    // If the label is on the right edge of the graph, text-align to the left
+    const textAlign = left === graphInfo.width + 5 ? "left" : "right";
+
+    // If the label requires padding for the negative sign, add it
+    // This allows us to ensure that labels are placed correctly when left-aligned
+    const negativeNumberPadding = label < 0 ? "12px" : "0px";
+
+    // Remove the negative sign from the label string as it is handled in the data-content attribute
+    const labelString = Math.abs(label).toString();
+
+    const styles = {
+        left,
+        top,
+        "text-align": textAlign,
+        "--padding": negativeNumberPadding,
+    };
+
+    return (
+        <>
+            {showTickLabel({
+                gridStep,
+                tickStep,
+                label,
+                graphEdge: graphInfo.height,
+                position: pixelPoint[1],
+            }) && (
+                <span
+                    className="y-axis-tick-label"
+                    style={styles}
+                    data-content={label < 0 ? "−" : null}
+                >
+                    {labelString}
+                </span>
+            )}
+        </>
+    );
+};
+
+const XGridLabel = ({label, gridStep, tickStep, graphInfo}: GridLabel) => {
+    const pointOnAxis: vec.Vector2 = [label, 0];
+    const pixelPoint = pointToPixel(pointOnAxis, graphInfo);
+
+    // If the label are outside of the graphs viewbox/range we want to make sure the
+    // labels are still visible by clamping them to the appropriate edge of the graph
+    const topPadding = graphInfo.range[1][1] < 0 ? -16 : 7;
```

Comment:
I think I should break these padding values out to named variables for clarity. I'm still not a fan of these hard-coded values but it makes a big difference for positioning depending on whether the axis is at the top or bottom of the graph.